src/main/java/teammates/storage/api/NotificationsDb.java

```diff
@@ -37,6 +42,61 @@ public NotificationAttributes getNotification(String notificationId) {
         return notificationId.isEmpty() ? null : makeAttributesOrNull(getNotificationEntity(notificationId));
     }
 
+    /**
+     * Gets all notifications.
+     */
+    public List<NotificationAttributes> getAllNotifications() {
+        List<Notification> notifications = load().list();
+        List<NotificationAttributes> notificationAttributes = makeAttributes(notifications);
+        NotificationAttributes.sortByStartTime(notificationAttributes);
+        return notificationAttributes;
+    }
+
+    /**
+     * Gets notifications by {@code targetUser}.
+     *
+     * @return a list of notifications for the specified targetUser.
+     */
+    public List<NotificationAttributes> getNotificationsByTargetUser(NotificationTargetUser targetUser) {
+        assert targetUser != null;
+
+        List<Notification> endEntities = load()
+                .filter("targetUser", targetUser)
+                .filter("endTime >", Instant.now())
+                .list();
+
+        List<Notification> startEntities = load()
+                .filter("targetUser", targetUser)
+                .filter("startTime <", Instant.now())
+                .list();
```

Comment:
why not use `.filter()` two times to filter the time? i.e.
```
        List<Notification> entities = load()
                .filter("targetUser", targetUser)
                .filter("startTime <", Instant.now())
                .filter("endTime >", Instant.now())
                .list();
```

Reply:
Datastore does not allow queries that uses two attributes inequality. Another example is `getAllOngoingSessions` in `FeedbackSessionsDb.java`.

Reply:
You made a point. Would it be ok if you add one composite index for this? I think it works after adding composite indices.

Because if you do query for two times, the cost will be many times higher especially when you have a large number of entities at a later stage. Even there is only one active notification, you will fetch almost everything from the database.

Reply:
Okay let me try that!

Reply:
Although multiple inequality filter is not supported for composite index, I have removed the query by `startTime` and use Java filter instead as suggested.